src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.Unix.props

```diff
@@ -14,10 +14,14 @@ The .NET Foundation licenses this file to you under the MIT license.
 -->
 <Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <PropertyGroup>
-    <CppCompilerAndLinker Condition="'$(CppCompilerAndLinker)' == ''">clang</CppCompilerAndLinker>
+    <CustomToolchainSpecified Condition="'$(CppCompilerAndLinker)' != ''">true</CustomToolchainSpecified>
+    <CppCompilerAndLinker Condition="'$(CustomToolchainSpecified)' != 'true'">clang</CppCompilerAndLinker>
+    <CppCompilerAndLinkerAlternative Condition="'$(CustomToolchainSpecified)' != 'true' and $(CppCompilerAndLinker.ToLowerInvariant().Contains('clang'))">gcc</CppCompilerAndLinkerAlternative>
```

Comment:
```suggestion
    <CppCompilerAndLinkerAlternative Condition="'$(CustomToolchainSpecified)' != 'true'">gcc</CppCompilerAndLinkerAlternative>
```

Reply:
CppCompilerAndLinker is always `clang` when `CustomToolchainSpecified` is not true.